### DISCUSSION TITLE
How to download models? Example GPT-J 6B

### DISCUSSION THREAD

**Discussion by metzo007** [oobabooga/text-generation-webui#243] [Q&A]
Hi,

there is more explanation needed in the "Downloading models". I can mace a bit of sence of if, but not enought to download a model.

First there is a Huggingface Link to gpt-j-6B. But I could not find any way to download the files from the page.

Second is says to use "python download-model.py organization/model" with the example "python download-model.py facebook/opt-1.3b". So I did try "python download-model.py EleutherAI/gpt-j-6B" but get a 

D:\oobabooga-windows\oobabooga\text-generation-webui>python download-model.py EleutherAI/gpt-j-6B
Traceback (most recent call last):
  File "D:\oobabooga-windows\oobabooga\text-generation-webui\download-model.py", line 17, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'

Error.

I dont know where to go from here. :-(

**Comment by wywywywy** [comment]
You need to activate the conda env first.

Alternatively, you can go to the `models` directory then do `git clone https://huggingface.co/EleutherAI/gpt-j-6B` assuming you have git installed and have git LFS enabled already.

To enable git LFS on your pc, simply do `git lfs install`

**Comment by JammersUK** [comment]
I've found the best way is to git clone a Hugging Face model, straight into the models directory, also.
I've just downloaded loads of different ones, just to see which work.  Not all of them are suitable, yet.
I have just tried to find out what a Pythia model does. Some very random and weird results.  Hugging Face says "Pythia models are English-language only,", I type in "Hola!", it's reply was "Hola amigo y querido chico usted tiene unos buenos días con su familia de la ciudad que se llama Barcelona o algo así como el País Bético en España???????? Bueno ya lo dije pero si...", not quite English only :-)  I've tried some QnA models also, which sort of work ok.  The Galactica models are quite interesting.  I had a chat about Set theory and Russels Paradox :-)  But most of the time, they crash or just end up giving loads of repeats.  So I'm guessing only a few are fully compatible for now.  Just download from Hugging Face and see what happens.

**Reply by henry808** [reply]
> I've found the best way is to git clone a Hugging Face model, straight into the models directory, also.

This is not the best way to do it, because it will pull in the whole repo and many of the model repos contain all kinds of extra files or may even have several models which tend to be multiple Gigs each.

Instead look for the exact model you need and download it directly. There should be a download button on most models in huggyface after you navigate to them.

**Comment by oobabooga** [comment]
GALACTICA is compatible, the problem is the gradio library that crashes when an invalid LaTeX equation is generated instead of ignoring that particular equation. Regenerating usually works in cases like this.

**Reply by JammersUK** [reply]
Oh, thank you.  I'm very new to this form of AI.  I started with Stable Diffusion.

**Comment by JammersUK** [comment]
Also watch out for file sizes.  I've found 2.7B/2.8B models are the maximum I can use on my 8GB VRAM card.  My card is only a RTX 2080 Super and running on Windows.

**Comment by oobabooga** [comment]
You can load larger models in 8-bit mode (see the links in the readme), and you can load llama-7b in 4-bit mode. It may be easier to do those things on WSL instead of on Windows itself.

**Reply by JammersUK** [reply]
I would love to get this computer to dual boot with Linux, but it has some boot mode protection to mess it about.  WSL isn't the best, but better than breaking a Windows install with dual booting.
